src/app/modules/timesheet/services/timesheet.service.ts

```diff
@@ -120,14 +120,44 @@ export class TimesheetService {
   }
 
   public openLastTimesheetInLocal(): boolean {
-    const timesheetsOfLocalStorage = this.getTimesheetsLocal();
+    const timesheetsOfLocalStorage = this.getLocalstorageTimesheetsList();
     if (timesheetsOfLocalStorage.length > 0) {
       this.setTimesheet(this._localSaveService.getLocalItem(timesheetsOfLocalStorage[timesheetsOfLocalStorage.length - 1]));
       return true;
     }
     return false;
   }
 
+  public getPresentTimesheet(transferedTimesheet: Timesheet) {
+    const timesheetsOfLocalStorage = this.getLocalstorageTimesheetsList();
+    const localStorageSize = this.getLocalstorageTimesheetsList().length;
+    for (let i = localStorageSize; i >= 0; i--) {
```

Comment:
`for (let LocalStorageTimesheet in LocalStorageTimesheetsList) { }` 
 it's better no ? And you don't need to use a size variable.

Reply:
If you want to have the latest CRA in first and use the `for ... in`, you can choose two options, add a method to sort the `timesheetList`, sort directly in the `getLocalstorageTimesheetsList` but you're probably going to have to change the name of the method.

Or you can do the sort directly here but the function is already long enough.

What do you think about that?